db/migrate/20211116115254_create_join_table_category_deal.rb

```diff
@@ -0,0 +1,8 @@
+class CreateJoinTableCategoryDeal < ActiveRecord::Migration[6.1]
+  def change
+    create_join_table :categories, :deals do |t|
+      t.index [:category_id, :deal_id]
+      # t.index [:deal_id, :category_id]
```

Comment:
- [ ] I would recommend removing the commented code to make the codebase cleaner.